crates/codegen/parser/runtime/src/support/choice_helper.rs

```diff
@@ -1,8 +1,12 @@
 use std::mem;
 use std::ops::ControlFlow;
 
-use super::{context::Marker, ParserContext, ParserResult};
-use crate::{cst, kinds::TokenKind, parse_error::ParseError, text_index::TextIndex};
+use super::context::Marker;
+use super::{ParserContext, ParserResult};
+use crate::cst;
+use crate::kinds::TokenKind;
```

Comment:
I would like to actually argue in favor of using grouping by `crate` rather than `module`, but I think this would be resolved easier in a f2f discussion 😅

Reply:
I prefer crate as well, because it gives me more immediate 'from which domain is this name'. I am used to it because I spend time doing it manually for precisely that reason. I prefer the visual structuring.

Reply:
Rarely the focus on the imported items should be on the crate, unless it's a library with <8-10ish items; often they're already (or should be) exported at the root or direct child level and then `use` looks similar in those cases already. Since modules are *the* grouping mechanism for items, I think it's more natural and readable to focus on that.

Importing items from multiple different modules in one line (e.g. [here](https://github.com/NomicFoundation/slang/pull/702/commits/a71e8dcbae1bb9daf446e4fe053bc8f7f336d190#diff-da04a52c132e4a848d34fe766e3e6b8cbac96b21d2ec84f3ec4e0188f6084746L10)) also makes it harder to distinguish different modules pulled at a glance.

Also, sorted lines of `module`-granular imports already visually form the tree, so that retains the visual cue that you mention. Moreover, in `crate`-granularity to understand a given line you have to keep track of the parent line repeatedly up to the root `use`, whereas with `module`-granularity you have all of the context/path in a single line, for each import.

I've given my reasoning here and in the OP https://github.com/NomicFoundation/slang/pull/702#issue-2035628402; I definitely respect democracy here :sweat_smile:  but I'd like for my arguments to be heard and also hear your arguments!

Reply:
f2f: let's try it out for now, and see how we like it.